src/App.php

```diff
@@ -19,7 +19,10 @@ public function __construct()
         $this->configureContainer($containerBuilder);
         $container = $containerBuilder->build();
 
-        parent::__construct($container);
+        parent::__construct($container->get('settings'));
+        $this->setContainer($container);
+        $this->getRouter()->setDefaultInvocationStrategy($container->get('foundHandler'));
+        $this->getRouter()->setCallableResolver($container->get('callableResolver'));
```

Comment:
These 2 lines could be called by PHP-DI: you can configure the injections on the router in https://github.com/PHP-DI/Slim-Bridge/blob/master/src/config.php#L39 directly. For example:

```php
    'router' => DI\object(Slim\Router::class)
        ->method('setCacheFile', DI\get('settings.routerCacheFile'))
        ->method('setDefaultInvocationStrategy', DI\get('foundHandler'))
        ->method('setCallableResolver', DI\get('callableResolver')),
```